src/middleware/operation.rs

```diff
@@ -280,11 +279,11 @@ impl Operation {
                         StatementArg::Key(ak2.clone()),
                     ])
                 } else {
-                    return Err(anyhow!("Invalid operation"));
+                    return Err(Error::Middleware(MiddlewareError::InvalidOp));
```

Comment:
is there a way to get automatic conversion from `MiddlewareError` to `Error` here?
So that we can write this:
```rust
return Err(MiddlewareError::InvalidOp);
```